bson/src/test/unit/org/bson/codecs/pojo/PojoCustomTest.java

```diff
@@ -176,6 +180,29 @@ public void apply(final ClassModelBuilder<?> classModelBuilder) {
                         + "           'simple_model': {'integer_field': 42, 'string_field': 'myString' } } }");
     }
 
+    @Test
+    public void testIdGeneratorMutable() {
```

Comment:
Are there already tests that cover the case where the instance already has an id set prior to encoding?  Just want to make sure that value is not ignored.